src/node_buffer.cc

```diff
@@ -19,7 +19,11 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Handle<Value> _obj = argT;                                                \
+  if (!HasInstance(_obj)) {                                                 \
+    return env->ThrowTypeError("must be called on Buffer instance");        \
```

Comment:
@bnoordhuis' approach is the correct one. when I first rewrote Buffer I did it that way because the diff was already huge and using `setupBufferJS` was less diff noise in the PR. Plus it maintained how things had been done before.

@bnoordhuis you have plans to rewrite Buffer, or want me to? actually, um. do we just want to hold off on this since Buffer has to be completely rewritten anyway since 4.4 will break everything?